lsp-mode.el

```diff
@@ -5203,18 +5239,44 @@ It will show up only if current point has signature help."
         :context `( :diagnostics ,(lsp-cur-line-diagnostics)
                     ,@(when kind (list :only (vector kind))))))
 
-(defun lsp-code-actions-at-point (&optional kind)
-  "Retrieve the code actions for the active region or the current line."
-  (lsp-request "textDocument/codeAction" (lsp--text-document-code-action-params kind)))
+(defun lsp-code-actions-at-point (&optional kind enabled)
+  "Retrieve the code actions for the active region or the current line.
+If ENABLED is specified, only return code actions that are not
+disabled."
+  (--doto (lsp-request "textDocument/codeAction" (lsp--text-document-code-action-params kind))
```

Comment:
I guess this section is on the edge of being too trickier.

Reply:
What I had in mind here was the following: currently there is a single filter: ENABLED. In the future, more filters could be added, and using `--doto` with `cl-callf` allows adding them easily (e.g. `is-preferred`). I could rewrite this in terms of `if` if you consider that to be a large enough readability gain:
```emacs-lisp
  (let ((actions (lsp-request "textDocument/codeAction" (lsp--text-document-code-action-params kind))))
    (if enabled
        (seq-remove #'lsp:code-action-disabled? actions)
      acitons))
```

I just realized that the code has a bug: `lsp-code-action-disabled?` should be `lsp:code-action-disabled?`.

Reply:
I guess it could stay as it is, just it is a bit unusual. Once `-cond->>` lands in `dash` we can use it here(and few other places).

Reply:
`-cond->>` would certainly be awesome, and I remember that I wanted to use it somewhere but couldn't. However, I don't see how this would help here? The point of this extensible function style is to allow filters to be chained: e.g. first filter out all disabled actions, then those that don't match the type base, .... (we don't have that yet, but might in the future).

Reply:
Something like that (clj)

```

(let [even true
      small true]
  (cond->> (range 20)
    even (filter even?)
    small (filter #(< 10 %))))
```